### PR TITLE
Migrate 'iframed masonry block' tests to Playwright

### LINKED ISSUE #34773: [Flaky Test] should load script and dependencies in iframe
**Flaky test detected. This is an auto-generated issue by GitHub Actions. Please do NOT edit this manually.**

## Test title
should load script and dependencies in iframe

## Test path
`specs/editor/plugins/iframed-masonry-block.test.js`

## Errors

<time datetime="2021-09-13T09:52:12.774Z"><code>[2021-09-13T09:52:12.774Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/1228853019"><code>trunk</code></a>.
<br/>
<time datetime="2021-09-13T11:07:20.775Z"><code>[2021-09-13T11:07:20.775Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/1229169706"><code>trunk</code></a>.
<br/>
<time datetime="2023-07-21T12:58:20.793Z"><code>[2023-07-21T12:58:20.793Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5621736371"><code>feat/async-block-loading</code></a>.
<br/>
<time datetime="2023-08-14T10:53:12.893Z"><code>[2023-08-14T10:53:12.893Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/5853131951"><code>feat/async-block-loading</code></a>.
<br/>
<time datetime="2023-09-11T13:53:37.361Z"><code>[2023-09-11T13:53:37.361Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6146965021"><code>try/iframe-skip-scoping-css</code></a>.
<br/>
<time datetime="2023-09-13T19:25:29.070Z"><code>[2023-09-13T19:25:29.070Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6176619602"><code>trunk</code></a>.
<br/>
<time datetime="2023-09-21T04:41:00.396Z"><code>[2023-09-21T04:41:00.396Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6257025010"><code>update/patterns-filter</code></a>.
<br/>
<time datetime="2023-09-26T11:03:12.214Z"><code>[2023-09-26T11:03:12.214Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6311437034"><code>update/ariakit</code></a>.
<br/>
<time datetime="2023-11-14T21:28:06.177Z"><code>[2023-11-14T21:28:06.177Z]</code></time> Test passed after 5 failed attempts on <a href="https://github.com/WordPress/gutenberg/actions/runs/6790573202"><code>experiment/show-styles-inherited-values</code></a>.
<br/>
<time datetime="2023-11-21T12:49:41.241Z"><code>[2023-11-21T12:49:41.241Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6942911498"><code>6-4-2-cherry-picks</code></a>.
<br/>
<time datetime="2023-11-21T13:11:20.281Z"><code>[2023-11-21T13:11:20.281Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6942911498"><code>6-4-2-cherry-picks</code></a>.
<br/>
<time datetime="2023-11-21T14:13:07.026Z"><code>[2023-11-21T14:13:07.026Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6942911498"><code>6-4-2-cherry-picks</code></a>.
<br/>
<time datetime="2023-11-21T14:47:26.492Z"><code>[2023-11-21T14:47:26.492Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6942911498"><code>6-4-2-cherry-picks</code></a>.
<br/>
<time datetime="2023-11-21T15:11:44.978Z"><code>[2023-11-21T15:11:44.978Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6942911498"><code>6-4-2-cherry-picks</code></a>.
<br/>
<time datetime="2023-11-21T15:46:20.953Z"><code>[2023-11-21T15:46:20.953Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/6942911498"><code>6-4-2-cherry-picks</code></a>.
<br/>
<time datetime="2023-11-27T03:52:33.506Z"><code>[2023-11-27T03:52:33.506Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/7000267194"><code>fix/template-replace-bug</code></a>.
<br/>
<details>
<summary>
	<time datetime="2023-11-27T04:09:53.531Z"><code>[2023-11-27T04:09:53.531Z]</code></time> Test passed after 1 failed attempt on <a href="https://github.com/WordPress/gutenberg/actions/runs/7000267194"><code>fix/template-replace-bug</code></a>.
</summary>

```
  ● iframed masonry block › should load script and dependencies in iframe

    TimeoutError: waiting for selector `tr[data-slug="gutenberg-test-iframed-masonry-block"] .deactivate a` failed: timeout 30000ms exceeded

      at new WaitTask (../../node_modules/puppeteer-core/src/common/DOMWorld.ts:813:28)
      at DOMWorld.waitForSelectorInPage (../../node_modules/puppeteer-core/src/common/DOMWorld.ts:656:22)
      at Object.internalHandler.waitFor (../../node_modules/puppeteer-core/src/common/QueryHandler.ts:78:19)
      at DOMWorld.waitForSelector (../../node_modules/puppeteer-core/src/common/DOMWorld.ts:511:25)
      at Frame.waitForSelector (../../node_modules/puppeteer-core/src/common/FrameManager.ts:1290:47)
      at Page.waitForSelector (../../node_modules/puppeteer-core/src/common/Page.ts:3222:29)
      at waitForSelector (../e2e-test-utils/build/@wordpress/e2e-test-utils/src/activate-plugin.js:29:13)
          at runMicrotasks (<anonymous>)
      at Object.<anonymous> (specs/editor/plugins/iframed-masonry-block.test.js:30:3)

```
</details>

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #38851.
Closes #34773.

PR migrates 'iframed masonry block' tests to Playwright and simplifies it. Now that the editor loads in iframed by default, extra steps are unnecessary.

## Why?
This is a part of general migration 🐦

## Testing Instructions
```
npm run test:e2e:playwright -- test/e2e/specs/editor/plugins/iframed-masonry-block.spec.js
```

### CODE REVIEW

File: test/e2e/specs/editor/plugins/iframed-masonry-block.spec.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+	test( 'should load script and dependencies in iframe', async ( {
+		editor,
+	} ) => {
+		await editor.insertBlock( { name: 'test/iframed-masonry-block' } );
+
+		const masonry = editor.canvas.getByRole( 'document', {
+			name: 'Block: Iframed Masonry Block',
+		} );
+		await expect( masonry ).toBeVisible();
```
Comment (Mamaduka) [inline test/e2e/specs/editor/plugins/iframed-masonry-block.spec.js:31]: I've replaced the snapshot assertion with this. Do we need to assert the block's whole edit markup?

**Review by ellatrix — APPROVED** [review]

File: test/e2e/specs/editor/plugins/iframed-masonry-block.spec.js (hunk 1 above)
Comment (kevin940726) [reply test/e2e/specs/editor/plugins/iframed-masonry-block.spec.js:31]: I don't think we do. If we want anything tested then we should probably add the assertions below anyway.

File: test/e2e/specs/editor/plugins/iframed-masonry-block.spec.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+		const masonryHeight = await masonry.evaluate( ( node ) =>
```
Comment (kevin940726) [inline test/e2e/specs/editor/plugins/iframed-masonry-block.spec.js]: Curious to know whether we can use `boundingBox` here?

**Review by kevin940726 — APPROVED** [review]
Another nice one 👍

File: test/e2e/specs/editor/plugins/iframed-masonry-block.spec.js (hunk 2 above)
Comment (Mamaduka) [reply test/e2e/specs/editor/plugins/iframed-masonry-block.spec.js]: Good idea! Updated in [9a66431](https://github.com/WordPress/gutenberg/pull/55016/commits/9a66431a2f755331fcef34bd1b9f0fe8428f1dcd).

**Pull request merged**